pysteps/nowcasts/steps.py

```diff
@@ -33,6 +33,873 @@
     DASK_IMPORTED = False
 
 
+class StepsNowcaster:
+    def __init__(self, precip, velocity, timesteps, **kwargs):
+        # Store inputs and optional parameters
+        self.precip = precip
+        self.velocity = velocity
+        self.timesteps = timesteps
+        self.n_ens_members = kwargs.get("n_ens_members", 24)
```

Comment:
In stead of using kwargs here, you could also make a `StepsNowcasterConfig`, which would be a dataclass. This way you can more explicitly specify to the user which additional arguments need to be provided and you could even specify type hints if you want.

Reply:
I will do that but would like to see your feedback on this at is is the first time I am using these type hints. Also @dnerini would this be something you would like to see in this version of pysteps? I know @mats-knmi is doing this for the v2 version but want to check if this is still within the pysteps 'style'

Reply:
This is done now! I might need to add typing and dataclass and typing as new optional dependicies if these are not in base python...